source/javascripts/components/StepItem/StepItem.tsx

```diff
@@ -10,24 +11,28 @@ import { Step } from '@/models';
 type StepItemProps = {
   workflowIndex: number;
   step: Step;
-  displayName: string;
-  version?: string;
   isSelected: boolean;
   hasVersionUpdate: boolean;
   onSelected: (step: Step, index: number) => void;
 };
 
 const tabIndex = (selected: boolean): number => (selected ? -1 : 0);
 
-const StepItem = ({
-  workflowIndex,
-  step,
-  displayName,
-  version,
-  hasVersionUpdate,
-  isSelected,
-  onSelected,
-}: StepItemProps): JSX.Element => {
+const StepItem = ({ workflowIndex, step, hasVersionUpdate, isSelected, onSelected }: StepItemProps): JSX.Element => {
+  const displayName = step.displayName();
+  const version = step.requestedVersion();
+
+  const isWithGroup = step.cvs === 'with';
```

Comment:
This should be a concern of the step to know this detail, to figure out which step is a with group.
Extend the step with `isWithGroup` and `isStepBundle` accordingly.